.github/workflows/osrm-backend.yml

```diff
@@ -396,7 +396,36 @@ jobs:
         elif [[ "${RUNNER_OS}" == "macOS" ]]; then
           echo "JOBS=$((`sysctl -n hw.ncpu` + 1))" >>  $GITHUB_ENV
         fi
+    # See: https://github.com/actions/toolkit/issues/946#issuecomment-1590016041
+    # We need it to be able to access system folders while restoring cached Boost below
+    - name: Give tar root ownership 
+      if: runner.os == 'Linux' && matrix.ENABLE_CONAN != 'ON'
+      run: sudo chown root /bin/tar && sudo chmod u+s /bin/tar
+    - name: Cache Boost
+      if: runner.os == 'Linux' && matrix.ENABLE_CONAN != 'ON'
+      id: cache-boost
+      uses: actions/cache@v4
+      with:
+        path: |
+          /usr/local/include/boost
+          /usr/local/lib/libboost*
+        key: v1-boost-${{ runner.os }}-${{ runner.arch }}-${{ matrix.runs-on }}
+        restore-keys: |
+          v1-boost-${{ runner.os }}-${{ runner.arch }}-${{ matrix.runs-on }}
 
+    - name: Install Boost
+      if: steps.cache-boost.outputs.cache-hit != 'true' && runner.os == 'Linux' && matrix.ENABLE_CONAN != 'ON'
+      run: |
+        BOOST_VERSION="1.85.0"
```

Comment:
There were problems with Boost from apt on Ubuntu 22.04, so I propose to install it manually for the time being.

Reply:
Would a switch to 24.04 make this easier?

Reply:
Well, with a switch to 24.04 I had another challenge: sol2 is not compiling with [newest clang we would have to switch to on 24.04](https://github.com/ThePhD/sol2/issues/1581). There is already fix in LLVM, so I believe it is just easier to wait for a while until this fix will come to GitHub Actions runner image for 24.04 and switch.

Reply:
The fix is only going to make it to clang 19, unfortunately.